### PR TITLE
[2072] Add a distinct second end-of-cycle chaser email

### PR DESCRIPTION
**Pull request by elceebee**
## Context

We send an email to candidates about the approaching Apply deadline. Originally, this was the same content for each message, but we will have distinct messages now. 

## Changes proposed in this pull request
- Added a second deadline reminder message. [Approved content for second reminder message](https://educationgovuk.sharepoint.com/:w:/r/sites/TeacherServices/_layouts/15/Doc.aspx?sourcedoc=%7B79FF7E19-A263-4C5C-B3F2-A1CA3031F726%7D&file=End%20of%20cycle%202024%20email%202.docx&action=default&mobileredirect=true)
- Ensured that the second reminder will be on a weekday. We might want to do the same for the first reminder, but this will all be reviewed when we do end of cycle next year.
- Updated related Email workers to enable sending either chaser type

## Guidance to review

locally or on the review app, the new message is: `/rails/mailers/candidate_mailer/eoc_second_deadline_reminder`

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [x] API release notes have been updated if necessary
- [x] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)
- [x] Inform data insights team due to database changes
- [x] Make sure all information from the Trello card is in here
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally
- [x] Add PR link to Trello card

### CODE REVIEW

**Comment by elceebee** [comment]
[Review and update content for scheduled email (19th Aug '24) to candidates about approaching Apply deadline](https://trello.com/c/qKOtMGlS/2072-review-and-update-content-for-scheduled-email-19th-aug-24-to-candidates-about-approaching-apply-deadline)

**Review by CatalinVoineag — APPROVED** [review]
Looks good to me, nice

File: app/services/send_eoc_deadline_reminder_email_to_candidate.rb [hunk 1]
```diff
@@ -1,17 +1,25 @@
 class SendEocDeadlineReminderEmailToCandidate
```
Comment (inulty-dfe) [inline app/services/send_eoc_deadline_reminder_email_to_candidate.rb:1]: Can you add a comment above the class declaration saying End Of Cycle for LSPs

**Review by inulty-dfe — APPROVED** [review]
Looked at the previews and looks good to me.

Only thing is we are using `eoc` in a lot of places, can we add a comment above the classes / methods so the documentation is picked up by LSPs etc? Non-blocking

**Review by dcyoung-dev — APPROVED** [review]
Nice work pulling this all together 🤩

File: app/services/send_eoc_deadline_reminder_email_to_candidate.rb (hunk 1 above)
Comment (elceebee) [reply app/services/send_eoc_deadline_reminder_email_to_candidate.rb:1]: I'm happy to do this, but I'm not sure exactly what you need. Is it literally just `# EndOfCycle` ?

**Comment by CatalinVoineag** [comment]
[Edit work experiences - Change db structure to accommodate work experience edits](https://trello.com/c/VzPy9EWy/2163-edit-work-experiences-change-db-structure-to-accommodate-work-experience-edits)

**Pull request merged**